alvr/server_openvr/cpp/alvr_server/Controller.cpp

```diff
@@ -284,18 +284,35 @@ bool Controller::onPoseUpdate(float predictionS, FfiHandData handData) {
         pose.vecPosition[1] = p[1];
         pose.vecPosition[2] = p[2];
 
-        pose.vecVelocity[0] = 0;
-        pose.vecVelocity[1] = 0;
-        pose.vecVelocity[2] = 0;
+        // If possible, use the last stored m_pose and timestamp
+        // to calculate the velocities of the current pose.
+        double linearVelocity[3] = { 0.0, 0.0, 0.0 };
+        vr::HmdVector3d_t angularVelocity = { 0.0, 0.0, 0.0 };
 
-        pose.vecAngularVelocity[0] = 0;
-        pose.vecAngularVelocity[1] = 0;
-        pose.vecAngularVelocity[2] = 0;
+        if (m_pose.poseIsValid) {
+            double dt = (double)(targetTimestampNs - m_poseTargetTimestampNs) / NS_PER_S;
```

Comment:
There is a chance targetTimestampNs might be smaller than m_poseTargetTimestampNs, so convert to double before the subtraction